### ISSUE TITLE
Request new version release - 0.4.9

### ISSUE THREAD

**Issue by lparsons** [joshuagryphon/plastid#24]
There were a few useful fixes made since 0.4.8 was released. I was hoping you could release an official version (0.4.9?) with the latest updates.

**Comment by joshuagryphon** [comment]
I'd be happy to. Originally I was hoping to close a few more open issues first, but I think it's obvious now that progress here has been slow as I have been swamped with a new[ish] job and other things. I could use some help maintaining plastid and keeping it up to date, if anybody would be up for volunteering.g

Before pushing this out, though, is there anything to merge in from #11 ?

**Comment by advance0921** [comment]
Hi Josh, I have been profoundly helped by Plastid and would be happy to maintain and keep it. I am not a sophisticated programmer yet but will be willing to learn!

**Comment by joshuagryphon** [comment]
Circling back to this- there are some updates I need to support newer versions of `pandas` as well as a few other packages; there are an awful lot of deprecation warnings when I run the unit tests. Please stay tuned- I will release when these are done (soon, because I'm working on them this week).

**Comment by lparsons** [comment]
BTW, if you know of version limits on `pandas` or other packages for `plastid 0.4.8` let me know and I'll update the `bioconda` recipe to reflect that...

**Comment by joshuagryphon** [comment]
@lparsons thank you! that would be great. I just ran the `plastid 0.4.8` tests, which passed with `pandas==0.19`. Other, more modern versions of everything else passed.

**Comment by joshuagryphon** [comment]
@lparsons `pandas>=0.17.0,<=0.19` is fine. The problem is that `pandas.DataFrame.sort()` (deprecated in v0.17.0) was finally removed in v0.20.0. This doesn't actually affect `plastid` proper, but does affect some of its tests.

I'm updating the tests, and hoping to also remove some other (now obsolete) code (e.g. from `setup.py`)

**Comment by joshuagryphon** [comment]
Update: all tests are now passing on both modern & historical versions of `pandas`. The last few things I'd like to do before bumping to 0.4.9 are to:

- wrap all of the `open()` statements in the tests inside context managers, to avoid creating stale filehandles
- simplify `setup.py` (or make it more complex), in order to avoid the annoying step that requires pre-installation of `numpy`, `pysam`, and `cython`

**Comment by joshuagryphon** [comment]
@lparsons great news- I have finished the remaining tasks above, including simplifying `setup.py`, and have been able to build & test in multiple environments (Py2.7, 2.5; Ubuntu 14.08 & 18.04). Would you be able to pull the latest `develop` commit and make sure it still installs under `conda`?

**Comment by lparsons** [comment]
@joshuagryphon I've been reminded of this, and will see what I can do. Hopefully we can get an update out soon (only a year! :wink: ).

**Comment by joshuagryphon** [comment]
tyvm! Vaporware always makes me sad. I still use this package occasionally, but not nearly as much as 2013-2017. Yes, let us get this out!

**Comment by joshuagryphon** [comment]
OK- one last thing I'd like to do before version bump – because it is probably going to be the last release for a while — is close #31 , which will enable compatibility with Python 3.7 (which is already here).

**Comment by sophiahjli** [comment]
@joshuagryphon It would be nice to have the compatibility! Any news for the version bump? Anything we can do to make it a reality? :)

**Comment by lparsons** [comment]
One last ping before we publish out workflow. Not a huge deal, we'll go ahead and use `0.4.8` if need be, but would be nice to have `0.4.9` released.

**Comment by joshuagryphon** [comment]
Hi folks,

Holiday travel has enabled me to resolve #31, which unblocks release of `0.4.9`. I will be able to push this early next week, once I dig get back home to my GPG key. Many thanks to you all for your patience.

@sophiahjli and @lparsons , thank you both for your offers of help in keeping development up. Given how little free time I have had for this project in the two years, I think it's a good time to come up with a plan to transition ownership of maintenance and development. I need to ping a few people about that, and will get back hopefully sooner than I have been.

I will close this issue when `0.4.9` has been pushed to PyPI

**Comment by lparsons** [comment]
bump

**Comment by lparsons** [comment]
Any chance you can find your GPG key these days and push to PyPi? Would be helpful for workflows to have the fixes.

**Comment by joshuagryphon** [comment]
Returning to this after ages- some good news:

* All changes from `develop` have been merged into `master`, and version 0.5.0 is now released here on GitHub (not yet pushed to PyPI- will do that in a bit)
* New version confirmed working under versions of Python all the way to 3.8, using latest version of all dependencies (e.g. `numpy`, `scipy`, `pysam` et c) as well as historical versions


*One warning if you have run a `apt-get upgrade` lately:*

There is something weird going on with compiling C extensions under recent versions of Ubuntu 18.04.

Concretely: everything compiles and runs on an old machine running 18.04 with GCC v7.4.0; everything compiles and breaks with an `undefined symbol` error (smells of C-compiler issues) on a new machine running a more up-to-date Ubuntu 18.04 with GCC v.7.5.0. All Python dependencies and Python install chain (pip, setuptools, et c) are identical between two machines, so the error is somewhere else.

Downgrading GCC did not help, so the problem is at least partially elsewhere. For details, see #41

**Comment by joshuagryphon** [comment]
Released to PyPI. Please enjoy everybody!
